src/lightning/fabric/strategies/xla_fsdp.py

```diff
@@ -0,0 +1,404 @@
+        return module.clip_grad_norm_(max_norm=max_norm, norm_type=norm_type)
+
+    def clip_gradients_value(  # type: ignore[override]
+        self, module: Module, optimizer: Optimizer, clip_val: Union[float, int]
+    ) -> None:
+        """Clip gradients by value."""
+        raise NotImplementedError(
+            "XLA's FSDP strategy does not support to clip gradients by value."
+            " Consider clipping by norm instead or choose another strategy!"
+        )
+
+    def save_checkpoint(
+        self,
+        path: _PATH,
+        state: Dict[str, Union[Module, Optimizer, Any]],
+        storage_options: Optional[Any] = None,
+        filter: Optional[Dict[str, Callable[[str, Any], bool]]] = None,
+    ) -> None:
+        """Save model, optimizer, and other state in a checkpoint directory.
+
+        The directory will contain one file per process, with model- and optimizer shards stored per file. Additionally,
+        it creates a a consolidated checkpoint combining all of the sharded checkpoints.
+        """
+        if not _TORCH_GREATER_EQUAL_2_0:
+            raise NotImplementedError(
+                "Saving and loading checkpoints with the `XLAFSDPStrategy` is not supported in PyTorch < 2.0."
+                " Please upgrade `torch` or file an issue: `https://github.com/Lightning-AI/lightning/issues`."
+            )
+        if storage_options is not None:
+            raise TypeError(
+                "`XLAFSDPStrategy.save_checkpoint(..., storage_options=...)` is not supported because"
+                " `XLAFSDPStrategy` does not use the `CheckpointIO`."
+            )
+        # broadcast the path from rank 0 to ensure all the states are saved in a common path
+        path = self.broadcast(path)
+        if Path(path).is_dir() and os.listdir(path):
+            raise FileExistsError(f"The checkpoint directory already exists and is not empty: {path}")
+        from torch_xla.distributed.fsdp import XlaFullyShardedDataParallel as XLAFSDP
+
+        modules = [module for module in state.values() if isinstance(module, XLAFSDP)]
+        if len(modules) == 0:
+            raise ValueError(
+                "Could not find a XLA FSDP model in the provided checkpoint state. Please provide the model as"
+                " part of the state like so: `save_checkpoint(..., state={'model': model, ...})`. Make sure"
+                " you set up the model (and optimizers if any) through the strategy before saving the checkpoint."
+            )
+        if len(modules) > 1:
+            raise ValueError(
+                "Found multiple XLA FSDP modules in the given state. Saving checkpoints with FSDP is"
+                " currently limited to a single model per checkpoint. To save multiple models, call the"
+                " save method for each model separately with a different path."
+            )
+        import torch_xla.core.xla_model as xm
+
+        rank = xm.get_ordinal()
+        world_size = xm.xrt_world_size()
```

Comment:
Try to always use the utilities that are available in the strategy

```suggestion
        rank = self.global_rank
        world_size = self.world_size
```